ui/packages/components/src/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+
+import { Button } from '../Button';
+import { Popover, PopoverClose, PopoverContent, PopoverTrigger } from '../Popover';
+import { Switch, SwitchLabel, SwitchWrapper } from '../Switch';
+import { combineDayAndTime, formatDayString, formatTimeString } from '../utils/date';
+import { Calendar } from './Calendar';
+import { DateInputButton } from './DateInputButton';
+import { TimeInput } from './TimeInput';
+
+type DatePickerProps = {
+  datePickerDate?: Date;
+  setDatePickerDate: React.Dispatch<React.SetStateAction<Date | undefined>>;
+};
+
+export function DatePicker({ datePickerDate, setDatePickerDate }: DatePickerProps) {
+  const [buttonCopy, setButtonCopy] = useState<string>(
+    datePickerDate?.toISOString() || 'Enter Date and Time'
+  );
```

Comment:
Since this is the data layer, it seems a little cleaner to store the datetime value. Down in the view layer, we could decide how we want to render the datetime value.

This approach also has the benefit of avoiding the "double formatting" code that currently exists: right now we're calling `.toISOString()` twice (once here and once later when calling `setButtonCopy`

```suggestion
export function DatePicker({ defaultValue, onChange }: DatePickerProps) {
  const [value, setValue] = useState<Date>(defaultValue);
```